src/state/queries/messages/list-converations.ts

```diff
@@ -27,3 +33,100 @@ export function useListConvos({refetchInterval}: {refetchInterval: number}) {
     refetchInterval,
   })
 }
+
+export function useUnreadMessageCount() {
+  const {currentConvoId} = useCurrentConvoId()
+  const convos = useListConvos({
+    refetchInterval: 30_000,
+  })
+
+  const count =
+    convos.data?.pages
+      .flatMap(page => page.convos)
+      .filter(convo => convo.id !== currentConvoId)
```

Comment:
This is optimistic, it immediate decrements the badge count when `currentConvoId` is set by the active chat screen.